pallets/crowdloan-rewards/src/lib.rs

```diff
@@ -51,124 +98,85 @@ pub mod pallet {
-			rate: u32,
-			end_block: T::BlockNumber,
+			contributions: Vec<(AccountIdOf<T>, BalanceOf<T>)>,
 		) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
-			let now = frame_system::Pallet::<T>::block_number();
-			ensure!(
-				&now < &end_block,
-				Error::<T>::InvalidEndBlock
-			);
-			let current_reward_end_in = CurrentRewardEndIn::<T>::get();
-
-			ensure!(
-				&now >= &current_reward_end_in,
-				Error::<T>::AlreadyInitReward
-			);
-			for (account, amount) in &contributions {
-				let reward_info = RewardInfo {
-					total_reward: BalanceOf::<T>::from(*amount)
-						.saturating_mul(BalanceOf::<T>::from(rate)),
-					claimed_reward: 0u32.into(),
-					last_paid: now.clone(),
-				};
-				Contributors::<T>::insert(account, reward_info);
-			};
-			CurrentRewardEndIn::<T>::put(&end_block);
-			RewardPeriod::<T>::put(end_block - now);
-			Ok(Default::default())
-		}
-
-		#[pallet::weight(0)]
-		pub fn associate_account(
-			origin: OriginFor<T>,
-			relay_account: T::RelayChainAccountId,
-			proof: MultiSignature,
-		) -> DispatchResultWithPostInfo {
-			let who = ensure_signed(origin)?;
-			let payload = who.clone().encode();
-
-			ensure!(
-				proof.verify(payload.as_slice(), &relay_account.clone().into()),
-				Error::<T>::InvalidSignature
-			);
-
-			ensure!(
-				AssociatedAccount::<T>::get(&who).is_none(),
-				Error::<T>::AlreadyAssociated
-			);
-
-			AssociatedAccount::<T>::insert(&who, &relay_account);
-
-			Self::deposit_event(Event::AssociatedAccount(
-				who,
-				relay_account,
-			));
-			Ok(Default::default())
-		}
-
-		#[pallet::weight(0)]
-		pub fn update_associate_account(
-			origin: OriginFor<T>,
-			associated_native_account: T::AccountId,
-			relay_account: T::RelayChainAccountId,
-			proof: MultiSignature,
-		) -> DispatchResultWithPostInfo {
-			let who = ensure_signed(origin)?;
-			let payload = who.clone().encode();
 
-			ensure!(
-				proof.verify(payload.as_slice(), &relay_account.clone().into()),
-				Error::<T>::InvalidSignature
-			);
-			ensure!(
-				AssociatedAccount::<T>::get(&who).is_none(),
-				Error::<T>::AlreadyAssociated
-			);
-			let relay_associated_account = AssociatedAccount::<T>::get(&associated_native_account)
-				.ok_or(Error::<T>::NoAssociatedAccount)?;
-
-			ensure!(
-				relay_account == relay_associated_account,
-				Error::<T>::BadRelayAccount
-			);
-			AssociatedAccount::<T>::remove(associated_native_account);
-			AssociatedAccount::<T>::insert(&who, &relay_account);
-			Self::deposit_event(Event::UpdateAssociateAccount(
-				who,
-				relay_account,
-			));
+			let relay_block_now = BlockNumberOf::<T>::from(cumulus_pallet_parachain_system::Pallet::<T>::validation_data()
+				.expect("set_validation_data inherent needs to be present in every block!")
+				.relay_parent_number);
+			let setting = Settings::<T>::get();
+			ensure!(InitRewardAt::<T>::get().is_zero(), Error::<T>::AlreadyInitReward);
+
+			let mut total_reward_amount = BalanceOf::<T>::from(0u32);
```

Comment:
```suggestion
		let total_reward_amount: BalanceOf<T> = contributions
			.iter()
			.fold(0u32.into(), |acc: BalanceOf<T>, (_, amount)| {
			acc + *amount
		});
```

Reply:
Changed it:
https://github.com/polkafoundry/newpolka/pull/55/commits/946b1f6b735d24efc3027a92072b587b9f3a83af#diff-2bbae40c9c7fdfeeb0b1708d368844303a595083bc948ccd684b577e5d13f6ebR111